static/js/public/vueComponents/browseDisplay.vue

```diff
@@ -0,0 +1,260 @@
+                    }
+
+                    return page_display;
+                },
+
+                numberOfRecords: function() {
+                    return this.record_list.length || 0;
+                }
+            },
+
+            methods: {
+                recordCountFormat: function(number) {
+                    return new Intl.NumberFormat().format(number);
+                },
+
+                recordType: function(type) {
+                    if (type === 'Collection') {
+                        return 'fa-archive';
+                    } else if (type === 'Folder') {
+                        return 'fa-folder';
+                    } else {
+                        return 'fa-file';
+                    }
+                },
+
+                numberOfColumns: function() {
+                    let screen_size = window.innerWidth;
+
+                    if (screen_size > 1023) {
+                        this.column_size = 'is-3';
+                    } else if (screen_size > 768) {
+                        this.column_size = 'is-4';
+                    } else {
+                        this.column_size = 'is-6';
+                    }
+                },
+
+                /**
+                 * Updates list with results of BrowseSearch component custom event
+                 * @param search_results
+                 */
+                browseSearching: function(search_results) {
+                    this.container_metadata = search_results.container;
+                    this.record_list = search_results.metadata;
+                },
+
+                /**
+                 * Updates list with results of BrowseSort component custom event
+                 * @param sorted_records
+                 */
+                sortOrdering: function(sorted_records) {
+                    this.record_list = sorted_records;
+                },
+
+                /**
+                 * Updates which page to show from the results of Pagination component custom event
+                 * @param pagination_settings
+                 */
+                pageToDisplay: function(pagination_settings) {
+                    this.pagination_settings = pagination_settings;
+                }
+            },
+
+            mounted: function() {
+                let self = this;
+                fetch(this.path)
+                    .then(function(response) {
+                        return response.json();
+                    }).then(function(data) {
+                        self.container_metadata = data.container;
+                        self.record_list = data.metadata;
+                    });
+
+                window.addEventListener('resize', _.debounce(this.numberOfColumns));
+            },
+
+            beforeDestroy: function() {
+                window.removeEventListener('resize', this.numberOfColumns);
+            }
+        });
+    });
+</script>
+
+<style>
+    .browse-records-display .columns {
```

Comment:
is it typical to put the css for a vue component into the vue file rather than in with the rest of the sites css? Seems like if we are using sass it would be easier to make things consistent if the css were centrally managed.

Reply:
Yeah, it usually is. In theory it can be scoped to just that component and you can use SASS in them. The whole CSS/HMTL in JS movement.